### PR TITLE
VEN-219 | Feature/refine logout

### PR DESCRIPTION
**Pull request by Zeukkari**

### CODE REVIEW

File: src/domain/pageHeader/PageHeaderContainer.tsx [hunk 1]
```diff
@@ -29,7 +29,15 @@ const PageHeaderContainer: React.SFC = () => {
       <Dropdown label={<Button icon={<Icon name="user" />}>{fullName}</Button>}>
         <List noBullets>
           <ListItem>
-            <Button variant="text" onClick={() => dispatch(userSignedOut())}>
+            <Button
+              variant="text"
+              onClick={() => {
+                // Flush data in redux store and localStorage
+                persistor.flush();
```
Comment (Nizar-Rahme) [inline src/domain/pageHeader/PageHeaderContainer.tsx]: Are we really persisting the data in the first place? I think we are not.

File: src/domain/auth/authenticate.ts [hunk 2]
```diff
@@ -50,6 +48,19 @@ export const authenticateWithBackend = (
 
     dispatch(fetchTokenSuccess(res.data));
   } catch (error) {
-    dispatch(fetchTokenError(error));
+    toast(i18n.t('authentication.errorMessage'), {
+      type: toast.TYPE.ERROR,
+    });
+    try {
+      // This is a workaround that can save us until we can fix silentRenew().
```
Comment (Nizar-Rahme) [inline src/domain/auth/authenticate.ts]: Would be better great if we create a ticket in the balcklog for this fix and mention it here, so it won't slip through the cracks.

File: package.json [hunk 3]
```diff
@@ -31,6 +31,7 @@
     "react-router-dom": "^5.1.2",
     "react-scripts": "^3.3.0",
     "react-table": "^7.0.0-rc.15",
+    "react-toastify": "^5.5.0",
```
Comment (Nizar-Rahme) [inline package.json]: I bet it is a temporary solution until we implement the notifications Joakim has designed, isn't it?

File: src/domain/auth/authenticate.ts [hunk 4]
```diff
@@ -50,6 +48,19 @@ export const authenticateWithBackend = (
 
     dispatch(fetchTokenSuccess(res.data));
   } catch (error) {
-    dispatch(fetchTokenError(error));
+    toast(i18n.t('authentication.errorMessage'), {
```
Comment (Nizar-Rahme) [inline src/domain/auth/authenticate.ts]: You forgot to add this error message to `fi.json` file.

File: src/domain/auth/authenticate.ts (hunk 2 above)
Comment (Zeukkari) [reply src/domain/auth/authenticate.ts]: we already have it - VEN-378

Comment (Nizar-Rahme) [reply src/domain/auth/authenticate.ts]: Cool, so can you mention it in the comment?

File: src/domain/auth/authenticate.ts (hunk 4 above)
Comment (Zeukkari) [reply src/domain/auth/authenticate.ts]: Fixed :)

File: package.json (hunk 3 above)
Comment (Zeukkari) [reply package.json]: Yeah, this is analogous to the spinning loader we added last week. Let's replace it with our own implementation when we have the time.

File: src/domain/pageHeader/PageHeaderContainer.tsx (hunk 1 above)
Comment (Zeukkari) [reply src/domain/pageHeader/PageHeaderContainer.tsx]: This is an intermediary solution which I would like to refactor in the long run. For now, we are using `redux-persist` package. I haven't properly tested this and would like to get rid of this all together.

Comment (Nizar-Rahme) [reply src/domain/pageHeader/PageHeaderContainer.tsx]: I looked into it, it's safe to say that we can get rid of it now (unless we opt to persist the login state). I believe this `redux-persist` has leaked to our code from another service, and it is not used at the moment.

File: package.json (hunk 3 above)
Comment (Nizar-Rahme) [reply package.json]: Fair enough.

File: src/domain/pageHeader/PageHeaderContainer.tsx (hunk 1 above)
Comment (Zeukkari) [reply src/domain/pageHeader/PageHeaderContainer.tsx]: ok

File: src/domain/auth/authenticate.ts (hunk 2 above)
Comment (Zeukkari) [reply src/domain/auth/authenticate.ts]: done

**Pull request merged**